File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.c
#include <math.h>
bool increasingTriplet(int* nums, int numsSize) {
    int i = 0, smallest = pow(2, 31) - 1, second_smallest = pow(2, 31) - 1;
    for(i = 0; i < numsSize; i++)
    {
        if(nums[i] <= smallest)
        {
            smallest = nums[i];
        }
        else if(nums[i] <= second_smallest)
        {
            second_smallest = nums[i];
        }
        else
        {
            return true;
        }
    }
    return false;
}

File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.py
INDEX_INDEX = 0
VALUE_INDEX = 1

class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        smallest, second_smallest = 2 ** 31 - 1, 2 ** 31 - 1
        for i in nums:
            if i <= smallest:
                smallest = i
            elif i <= second_smallest:
                second_smallest = i
            else:
                return True
        return False
        